update `hugo` cli to `0.126.x`

With the currently latest version of `hugo` cli, I get following error messages:

```bash
make serve
./scripts/serve.sh
error: [401] SAML SSO login is required.
Hugo provides its own webserver which builds and serves the site.
While hugo server is high performance, it is a webserver with limited options.

The `hugo server` command will by default write and serve files from disk, but
you can render to memory by using the `--renderToMemory` flag. This can be
faster in some cases, but it will consume more memory.

By default hugo will also watch your files for any changes you make and
automatically rebuild the site. It will then live reload any open browser pages
and push the latest content to them. As most Hugo sites are built in a fraction
of a second, you will be able to save and see your changes nearly instantly.

Usage:
  hugo server [command] [flags]
  hugo server [command]

Aliases:
  server, serve

Available Commands:
  trust       Install the local CA in the system trust store.

Flags:
      --appendPort               append port to baseURL (default true)
  -b, --baseURL string           hostname (and path) to the root, e.g. https://spf13.com/
      --bind string              interface to which the server will bind (default "127.0.0.1")
  -D, --buildDrafts              include content marked as draft
  -E, --buildExpired             include expired content
  -F, --buildFuture              include content with publishdate in the future
      --cacheDir string          filesystem path to cache directory
      --cleanDestinationDir      remove files from destination not found in static directories
  -c, --contentDir string        filesystem path to content directory
      --disableBrowserError      do not show build errors in the browser
      --disableFastRender        enables full re-renders on changes
      --disableKinds strings     disable different kind of pages (home, RSS etc.)
      --disableLiveReload        watch without enabling live browser reload on rebuild
      --enableGitInfo            add Git revision, date, author, and CODEOWNERS info to the pages
      --forceSyncStatic          copy all files when static is changed.
      --gc                       enable to run some cleanup tasks (remove unused cache files) after the build
  -h, --help                     help for server
      --ignoreCache              ignores the cache directory
  -l, --layoutDir string         filesystem path to layout directory
      --liveReloadPort int       port for live reloading (i.e. 443 in HTTPS proxy situations) (default -1)
      --minify                   minify any supported output format (HTML, XML etc.)
      --navigateToChanged        navigate to changed content file on live browser reload
      --noBuildLock              don't create .hugo_build.lock file
      --noChmod                  don't sync permission mode of files
      --noHTTPCache              prevent HTTP caching
      --noTimes                  don't sync modification time of files
      --panicOnWarning           panic on first WARNING log
      --poll string              set this to a poll interval, e.g --poll 700ms, to use a poll based approach to watch for file system changes
  -p, --port int                 port on which the server will listen (default 1313)
      --pprof                    enable the pprof server (port 8080)
      --printI18nWarnings        print missing translations
      --printMemoryUsage         print memory usage to screen at intervals
      --printPathWarnings        print warnings on duplicate target paths etc.
      --printUnusedTemplates     print warnings on unused templates.
      --renderSegments strings   named segments to render (configured in the segments config)
      --renderStaticToDisk       serve static files from disk and dynamic files from memory
      --templateMetrics          display metrics about template executions
      --templateMetricsHints     calculate some improvement hints when combined with --templateMetrics
  -t, --theme strings            themes to use (located in /themes/THEMENAME/)
      --tlsAuto                  generate and use locally-trusted certificates.
      --tlsCertFile string       path to TLS certificate file
      --tlsKeyFile string        path to TLS key file
      --trace file               write trace to file (not useful in general)
  -w, --watch                    watch filesystem for changes and recreate as needed (default true)

Global Flags:
      --clock string               set the clock used by Hugo, e.g. --clock 2021-11-06T22:30:00.00+09:00
      --config string              config file (default is hugo.yaml|json|toml)
      --configDir string           config dir (default "config")
      --debug                      debug output
  -d, --destination string         filesystem path to write files to
  -e, --environment string         build environment
      --ignoreVendorPaths string   ignores any _vendor for module paths matching the given Glob pattern
      --logLevel string            log level (debug|info|warn|error)
      --quiet                      build in quiet mode
      --renderToMemory             render to memory (mostly useful when running the server)
  -s, --source string              filesystem path to read files relative from
      --themesDir string           filesystem path to themes directory
  -v, --verbose                    verbose output

Use "hugo server [command] --help" for more information about a command.

Error: command error: unknown flag: --renderToDisk
make: *** [serve] Error 1
```

The flag is called in my hugo -> renderStaticToDisk

```bash
Error: error building site: render: failed to render pages: render of "home" failed: "/Users/dirien/Tools/repos/docs/layouts/index.json:12:32": execute of template failed: template: index.json:12:32: executing "index.json" – File is nil; wrap it in if or with: {{ with .File }}{{ .UniqueID }}{{ end }}
make: *** [serve] Error 1
```

--- a/.github/workflows/build-and-deploy.yml
+++ b/.github/workflows/build-and-deploy.yml
@@ -27,7 +27,7 @@ jobs:
 
       - uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
 
       - name: Configure AWS Credentials

--- a/.github/workflows/check-links.yml
+++ b/.github/workflows/check-links.yml
@@ -20,7 +20,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
       - name: Run make check_links
         run: make check_links

--- a/.github/workflows/customer-managed-deployment-agent-cli.yml
+++ b/.github/workflows/customer-managed-deployment-agent-cli.yml
@@ -54,7 +54,7 @@ jobs:
 #      - name: Install Hugo
 #        uses: peaceiris/actions-hugo@v2
 #        with:
-#          hugo-version: '0.111.0'
+#          hugo-version: '0.126.0'
 #          extended: true
 #      - name: Install node
 #        uses: actions/setup-node@v2

--- a/.github/workflows/esc-cli.yml
+++ b/.github/workflows/esc-cli.yml
@@ -53,7 +53,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
       - name: Install node
         uses: actions/setup-node@v2

--- a/.github/workflows/generate-provider-docs.yml
+++ b/.github/workflows/generate-provider-docs.yml
@@ -61,7 +61,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
       - name: setup node
         uses: actions/setup-node@v2

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -23,7 +23,7 @@ jobs:
 
       - uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
 
       - name: Configure AWS Credentials

--- a/.github/workflows/pulumi-cli.yml
+++ b/.github/workflows/pulumi-cli.yml
@@ -56,7 +56,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
       - name: Install node
         uses: actions/setup-node@v2

--- a/.github/workflows/scheduled-test.yml
+++ b/.github/workflows/scheduled-test.yml
@@ -57,7 +57,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: "0.111.0"
+          hugo-version: "0.126.0"
           extended: true
 
       - name: Install Pulumi

--- a/.github/workflows/scheduled-upgrade-programs.yml
+++ b/.github/workflows/scheduled-upgrade-programs.yml
@@ -57,7 +57,7 @@ jobs:
       - name: Install Hugo
         uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: "0.111.0"
+          hugo-version: "0.126.0"
           extended: true
 
       - name: Install Pulumi

--- a/.github/workflows/testing-build-and-deploy.yml
+++ b/.github/workflows/testing-build-and-deploy.yml
@@ -27,7 +27,7 @@ jobs:
 
       - uses: peaceiris/actions-hugo@v2
         with:
-          hugo-version: '0.111.0'
+          hugo-version: '0.126.0'
           extended: true
 
       - name: Configure AWS Credentials

--- a/.husky/pre-commit
+++ b/.husky/pre-commit
@@ -1,0 +1,2 @@
+echo "Running pre-commit checks..."
+make lint

--- a/Makefile
+++ b/Makefile
@@ -142,4 +142,4 @@ lint:
 
 .PHONY: format
 format:
-	yarn prettier --write .
+	./scripts/format.sh

--- a/README.md
+++ b/README.md
@@ -25,10 +25,10 @@ See also:
 
 We build the Pulumi website with Hugo, manage our dependencies with Node.js and Yarn, and write our documentation in Markdown. Below is a list of the tools you'll need if you'd like to work on the website (e.g., to contribute docs content, a blog post, etc.):
 
-* [Hugo](https://gohugo.io) (>= 0.111.0)
-  * Hugo 0.111.0 is highly recommended. This is the version we use in our deployment pipelines.
-* [Node.js](https://nodejs.org/en/) (>= 18)
-* [Yarn](https://classic.yarnpkg.com/en/) (1.x)
+* [Hugo](https://gohugo.io/installation/) (>= 0.126.0)
+  * Hugo 0.126.0 is highly recommended. This is the version we use in our deployment pipelines.
+* [Node.js](https://nodejs.org/en/download/package-manager) (>= 18)
+* [Yarn](https://classic.yarnpkg.com/lang/en/docs/install) (1.x)
 
 Additionally, to build the SDK and CLI documentation, you'll also need:
 
@@ -65,12 +65,12 @@ The `Makefile` exposes a number of useful helpers for authoring:
 * `make new-blog-post` scaffolds a new, bare-bones blog post with placeholder content
 * `make new-example-program` generates a new multi-language set of examples at `./static/programs`
 
-As a content contributor, the commands you'll most often use are these:
+As a content contributor, the commands you'll use most often are these:
 
 ```bash
 make ensure    # Install or update dependencies.
 make serve     # Run the development server locally on http://localhost:1313.
-make lint      # Identify any Markdown or code-formatting issues so you an fix them.
+make lint      # Identify any Markdown or code-formatting issues so you can fix them.
 ```
 
 ## Generating SDK and CLI documentation

--- a/config/_default/config.yml
+++ b/config/_default/config.yml
@@ -13,8 +13,7 @@ security:
       - ALGOLIA_APP_SEARCH_KEY
 
 disableKinds:
-  - category
-  - taxonomyTerm
+  - taxonomy
 
 sectionPagesMenu: main
 pygmentsCodeFences: true

--- a/layouts/index.json
+++ b/layouts/index.json
@@ -6,17 +6,19 @@
         {{ $ancestors = $ancestors | append $a.LinkTitle }}
     {{- end -}}
 
-    {{-
-        $index = $index | append (
-            dict
-                "objectID" $page.File.UniqueID
-                "title" $page.Title
-                "section" $page.Section
-                "href" $page.RelPermalink
-                "params" $page.Params
-                "kind" $page.Kind
-                "ancestors" (after 1 $ancestors)
-        )
-    -}}
+    {{- if ne $page.File nil -}}
+        {{-
+            $index = $index | append (
+                dict
+                    "objectID" $page.File.UniqueID
+                    "title" $page.Title
+                    "section" $page.Section
+                    "href" $page.RelPermalink
+                    "params" $page.Params
+                    "kind" $page.Kind
+                    "ancestors" (after 1 $ancestors)
+            )
+        -}}
+    {{- end -}}
 {{- end -}}
 {{- $index | jsonify (dict "indent" "    ") -}}

--- a/layouts/page/about.html
+++ b/layouts/page/about.html
@@ -57,7 +57,7 @@
         <h3 class="font-medium text-6xl">Our purpose</h3>
         <h4 class="font-medium text-4-xl">To democratize the cloud for every engineer.</h4>
         <p class="text-gray-700 text-xl mb-10">We help engineers ship infrastructure faster with Infrastructure as Code in general-purpose languages.</p>
-        <a href="{{ relref . " /leadership" }}" class="btn-primary">Meet our leaders</a>
+        <a href="{{ relref . "/leadership" }}" class="btn-primary">Meet our leaders</a>
     </section>
 
     <section id="what-we-believe" class="text-center text-lg py-8 relative">
@@ -98,7 +98,7 @@
             </div>
         </div>
         <div class="careers-cta py-6 w-full mt-28 bg-white">
-            <h4>Like what we stand for? <a href="{{ relref . " /careers" }}" class="font-bold text-blue-600">Join us.</a></h4>
+            <h4>Like what we stand for? <a href="{{ relref . "/careers" }}" class="font-bold text-blue-600">Join us.</a></h4>
             <p>We are looking for new teammates who want to build the future of cloud infrastructure together.</p>
         </div>
     </section>
@@ -119,7 +119,7 @@
                 <img src="/images/mascot/victory-medal-pulumipus.svg" alt="Pulumipus wearing a medal with two hands raised" />
                 <div class="header mt-6 mb-1.5 text-medium text-2xl">The Puluminaries</div>
                 <p class="text-left flex-grow">A group for stellar Pulumi community members who have made "above and beyond" contributions to the community.</p>
-                <a href="{{ relref . " /community/puluminaries" }}" class="btn-primary">Meet our Puluminaries</a>
+                <a href="{{ relref . "/community/puluminaries" }}" class="btn-primary">Meet our Puluminaries</a>
             </div>
             <div class="community-spotlight mx-auto flex flex-col justify-cetner items-center w-72 md:w-1/3 my-6 px-3">
                 <img src="/images/mascot/rocketpus.svg" alt="Pulumipus wearing a medal with two hands raised" />
@@ -129,9 +129,9 @@
             </div>
             <div class="community-spotlight mx-auto flex flex-col justify-cetner items-center w-72 md:w-1/3 my-6 px-3">
                 <img src="/images/mascot/hoodie-pulumipus.svg" alt="Pulumipus wearing a Pulumi hoodie" />
-                <div class="header mt-6 mb-1.5 text-medium text-2xl">Pulumi users</div>
+                <div class="header  /leadershipt-6 mb-1.5 text-medium text-2xl">Pulumi users</div>
                 <p class="text-left flex-grow">Engineers love Pulumi! Don’t believe us? Check out what our users have to say.</p>
-                <a href="{{ relref . " /testimonials" }}" class="btn-primary ">Read user quotes</a>
+                <a href="{{ relref . "/testimonials" }}" class="btn-primary ">Read user quotes</a>
             </div>
         </div>
     </section>
@@ -226,7 +226,7 @@
                 {{ end }}
             </ul>
             <div>
-                <a class="inline-block mt-6 text-bold text-blue-600 mt-12 font-bold" href="{{ relref . " /awards" }}">View all achievements</a>
+                <a class="inline-block mt-6 text-bold text-blue-600 mt-12 font-bold" href="{{ relref . "/awards" }}">View all achievements</a>
             </div>
         </div>
     </section>
@@ -248,7 +248,7 @@
                         </li>
                     {{ end }}
                 </ul>
-                <a class="text-blue-600 font-bold" href="{{ relref . " /newsroom" }}">View all press releases</a>
+                <a class="text-blue-600 font-bold" href="{{ relref . "/newsroom" }}">View all press releases</a>
             </div>
             <div class="md:w/1-2 md:px-8">
                 <h3 class="text-center md:text-left font-medium text-6xl">Recent News</h3>
@@ -262,7 +262,7 @@
                         </li>
                     {{ end }}
                 </ul>
-                <a class="text-blue-600 font-bold" href="{{ relref . " /newsroom" }}">View all recent news</a>
+                <a class="text-blue-600 font-bold" href="{{ relref . "/newsroom" }}">View all recent news</a>
             </div>
         </div>
     </section>
@@ -275,6 +275,6 @@
         <h3 class="font-medium text-6xl">Join our team</h3>
         <h4>We are fully remote and hiring!</h4>
         <p class="mb-9">Our team is a diverse and talented group of individuals from all walks of life.</p>
-        <a href="{{ relref . " /careers" }}" class="btn-primary">Browse open positions</a>
+        <a href="{{ relref . "/careers" }}" class="btn-primary">Browse open positions</a>
     </section>
 {{ end }}

--- a/layouts/page/leadership.html
+++ b/layouts/page/leadership.html
@@ -4,7 +4,7 @@
 
 {{ define "main" }}
     <div class="container mx-auto mt-12  max-w-7xl">
-        <a class="text-blue-600 text-xl font-body ml-9 hidden sm:block" href="{{ relref . " /about" }}">
+        <a class="text-blue-600 text-xl font-body ml-9 hidden sm:block" href="{{ relref . "/about" }}">
             <i class="fas fa-arrow-left mr-4"></i>
             About
         </a>

--- a/layouts/product/pulumi-cloud.html
+++ b/layouts/product/pulumi-cloud.html
@@ -151,7 +151,7 @@
             <h2>{{ .title }}</h2>
             <p class="max-w-xl mx-auto">{{ .description | markdownify }}</p>
             <div class="mt-12">
-                <a href="{{ relref $pageContext " /pricing" }}" class="btn-secondary">Pricing</a>
+                <a href="{{ relref $pageContext "/pricing" }}" class="btn-secondary">Pricing</a>
             </div>
         {{ end }}
     </section>

--- a/layouts/product/pulumi-insights.html
+++ b/layouts/product/pulumi-insights.html
@@ -120,7 +120,7 @@
             <h2>{{ .title }}</h2>
             <p class="max-w-2xl mx-auto">{{ .description | markdownify }}</p>
             <div class="mt-12">
-                <a href="{{ relref $pageContext " /pricing" }}" class="btn-secondary">View Pricing</a>
+                <a href="{{ relref $pageContext "/pricing" }}" class="btn-secondary">View Pricing</a>
             </div>
         {{ end }}
     </section>

--- a/package.json
+++ b/package.json
@@ -4,7 +4,8 @@
   "private": true,
   "license": "Apache-2.0",
   "scripts": {
-    "minify-css": "node scripts/minify-css.js"
+    "minify-css": "node scripts/minify-css.js",
+    "prepare": "husky"
   },
   "dependencies": {
     "@aws-sdk/client-s3": "^3.427.0",
@@ -34,6 +35,7 @@
     "typescript": "^4.9.5"
   },
   "devDependencies": {
+    "husky": "^9.0.11",
     "prettier": "^2.6.2"
   }
 }

--- a/scripts/ensure.sh
+++ b/scripts/ensure.sh
@@ -1,14 +1,34 @@
 #!/bin/bash
 
-# Make sure the Node version matches what's in .nvmrc (and in our GitHub Actions workflows).
-current_version="$(node -v)"
-required_version="$(cat .nvmrc)"
+set -o errexit -o pipefail
 
-if [ ! $(echo ${current_version} | grep ${required_version}) ]; then
-    printf "\nIt looks like you're running Node %s, but this project uses Node %s.\n" ${current_version} ${required_version}
-    printf "If you're using nvm, try running nvm use.\n\n"
-    exit 1
-fi
+check_version() {
+    tool_name="$1"
+    executable="$2"
+    version_string="$(eval $3)"
+    required_version="$4"
+    level="$5"
+
+    color_red=$(echo -e "\033[0;31m")
+    color_yellow=$(echo -e "\033[0;33m")
+    color_end=$(echo -e "\033[0m")
+
+    details="See the README at https://github.com/pulumi/docs for a list of required tools and versions."
+
+    if ! command -v "$executable" &> /dev/null; then
+        echo -e "${color_red}error${color_end} This project requires $1, but the '$2' executable is either not installed or not on your PATH."
+        echo $details
+        exit 1
+    fi
+
+    if [ ! $(echo ${version_string} | grep ${required_version}) ]; then
+        printf "${color_yellow}warning${color_end} It looks like you're running %s %s, but this project uses version %s.\n" ${tool_name} ${version_string} ${required_version}
+        echo $details
+    fi
+}
+check_version "Node.js"  "node"  "node -v | sed 's/v\([0-9\.]*\).*$/\1/'"            "18"
+check_version "Hugo"     "hugo"  "hugo version | sed 's/hugo v\([0-9\.]*\).*$/\1/'"  "0.126.0"
+check_version "Yarn"     "yarn"  "yarn -v | sed 's/v\([0-9\.]*\).*$/\1/'"            "1.22"
 
 # Install the Node dependencies for the website and the infrastructure.
 yarn install

--- a/scripts/format.sh
+++ b/scripts/format.sh
@@ -1,0 +1,5 @@
+#!/bin/bash
+
+set -o errexit -o pipefail
+
+yarn prettier --write .

